### PR TITLE
DOCKER-618: check for invalid cloudapi URL

### PR DESCRIPTION
**Pull request by misterdjules**
When an invalid cloudapi URL is entered, an error message is displayed
and the user is prompted to enter another URL.

This uses bash arrays, I'm not sure if that's a problem since we explicitly mention bash in the shebang.

### CODE REVIEW

**Comment by trentm** [comment]
@misterdjules sdc-docker-setup.sh is meant to work for non-JPC standups. Does this break that?

**Comment by misterdjules** [comment]
@trentm Excellent point, updated this PR to use the CloudAPI `/--ping` endpoint instead.

File: tools/sdc-docker-setup.sh [hunk 1]
```diff
@@ -332,30 +357,52 @@ shift $((OPTIND - 1))
...
+        cloudapiUrl=https://$cloudapiUrl.api.joyent.com
+    elif [[ "${cloudapiUrl:0:8}" != "https://" ]]; then
+        cloudapiUrl=https://$cloudapiUrl
+    fi
+    debug "cloudapiUrl: $cloudapiUrl"
+
+    cloudApiUrlStatus=$(pingCloudAPIUrl "$cloudapiUrl")
+    if [[ "$cloudApiUrlStatus" == "maintenance" ]]; then
+        printf "\"$cloudapiUrl\" seems to be in maintenance, please try another SDC CloudAPI URL\n"
+    elif [[ "$cloudApiUrlStatus" == "unavailable" ]]; then
```
Comment (trentm) [inline tools/sdc-docker-setup.sh]: I'd drop the "seems" unsure language. Perhaps:  "'$url' is current in maintenance. Please try again later\n"

**Comment by trentm** [comment]
@misterdjules You also changed the mode on tools/sdc-docker-setup.sh unintentionally?

File: tools/sdc-docker-setup.sh [hunk 2]
```diff
@@ -332,30 +357,52 @@ shift $((OPTIND - 1))
...
+        printf "Cannot ping \"$cloudapiUrl\", are you sure it is a valid SDC CloudAPI URL?\n"
+    elif [[ "$cloudApiUrlStatus" == "available" ]]; then
+        break
+    else
+        printf "There was an error when checking $cloudapiUrl's status\n"
+    fi
```
Comment (trentm) [inline tools/sdc-docker-setup.sh]: Perhaps a `fatal` if hitting this branch? Thoughts?

**Comment by trentm** [comment]
Otherwise LGTM. Thanks!

File: tools/sdc-docker-setup.sh (hunk 1 above)
Comment (misterdjules) [inline tools/sdc-docker-setup.sh]: Sounds good, will fix, thanks!

File: tools/sdc-docker-setup.sh (hunk 2 above)
Comment (misterdjules) [inline tools/sdc-docker-setup.sh]: Sounds good, fixed and updated the PR.

**Comment by misterdjules** [comment]
@trentm Good catch for the changed mode, it wasn't intentional. Updated this PR to address all your comments, will land soon. Thank you!

**Comment by misterdjules** [comment]
Landed in d429d88d71aa819c2d965d4af4f58c0f7952eb99.

**Pull request closed without merging**